flink-core/src/main/java/org/apache/flink/api/common/restartstrategy/RestartStrategies.java

```diff
@@ -72,9 +72,23 @@ public static RestartStrategyConfiguration fixedDelayRestart(
      * @param restartAttempts Number of restart attempts for the FixedDelayRestartStrategy
      * @param delayInterval Delay in-between restart attempts for the FixedDelayRestartStrategy
      * @return FixedDelayRestartStrategy
+     * @deprecated Use {@link #fixedDelayRestart(int, Duration)}
      */
+    @Deprecated
```

Comment:
IIUC, we don't need to change this class. 

The whole `RestartStrategies` class has been marked as `Deprecated` in the master branch.

Reply:
That's correct. This change can be reverted. :+1:

Reply:
I went through the change once more and came to the conclusion that we could keep this change. You're right when stating that the `RestartStrategies` are going away, anyway. But it doesn't hurt that we do the deprecation "independently" of the `RestartStrategies` efforts. That way, we would still have a consistent code base if (for whatever reason) [FLIP-381](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=278464992) gets reverted. WDYT?

Reply:
You are right, this change is useful if FLIP-381 is reverted. But it's very unlikely. And these changes can be done when it happens.

So far, this change isn't required, and I'm fine to have it. So up to you.

Reply:
Another reason would be that not touching `RestartStrategies` means that we have a (soft) dependencies between FLIP-381 and FLIP-335. With the alternative methods in `RestartStrategies` in place, we can go ahead with FLIP-335 in a more straight-forward fashion and don't need to wait with removing the `Time` classes until  FLIP-381 is resolved. :thinking:

Reply:
Thanks for the clarifying. LGTM